packages/react-instantsearch-hooks/src/components/InstantSearchSSRProvider.tsx

```diff
@@ -18,10 +18,14 @@ export type InstantSearchSSRProviderProps =
  * Provider to pass the server state retrieved from `getServerState()` to
  * <InstantSearch>.
  */
-export function InstantSearchSSRProvider({
-  children,
-  ...props
-}: InstantSearchSSRProviderProps) {
+export function InstantSearchSSRProvider<
+  TUiState extends UiState,
+  TRouteState = TUiState
+>({ children, ...props }: InstantSearchSSRProviderProps) {
+  const ssrSearchRef = React.useRef<InstantSearch<UiState, TRouteState> | null>(
```

Comment:
is the `notify` callback as used for getRenderState not sufficient?

Reply:
The problem is client-side only, after `getServerState` did its thing so that's why I only handle it with `InstantSearchSSRContext`.
Do you mean we should handle it for both `getServerState` rendering and Next.js SSR ? I put the presence of `InstantSearchSSRContext` to be the condition to use the ref from this context so I guess it should work.

Reply:
Ah sorry you meant to merge the two contexts and just use `notifyServer` instead ?
I'll see if it's possible yeah